cli/pub_command.go

```diff
@@ -403,6 +490,13 @@ func (c *pubCmd) publish(_ *fisk.ParseContext) error {
 						eof = true
 					}
 					if body == "" && eof {
+						if c.atomic {
```

Comment:
seems a bit nasty to put it here, we could add a new `if c.atomic` above line 525 and do all the atomic stuff there rather than squeeze it into the jetstream check and worse into `doJetstream`, doing it there in its own func means we can just put all the atomic things together and branch in that function around `eof` being true?

Reply:
Moving the if atomic check out of the jetstream check is a good idea. Should have cleaned this up when I changed approach.

Annoyingly the `commit` has to happen on EOF or in the select. I'm going to move it to the select because I think it's going to be less nested noise, and let's see how it looks.